app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,146 @@
+module Authentication
+  module AuthnJwt
+    module VendorConfigurations
+      # Mock JWTConfiguration class to use it to develop other part in the jwt authenticator
+      class ConfigurationJWTGenericVendor < ConfigurationInterface
```

Comment:
Instead of using inheritance to hold shared functionality, how about using composition? Shared functionality can be passed into the initialization method and used by the receiving class. The external class is easily tested and it's a snap and to mock the dependency.

Reply:
This is not relevant anymore